apps/web/pages/signup.tsx

```diff
@@ -255,14 +256,35 @@ export default function Signup({
           pushGTMEvent("create_account", { email: data.email, user: data.username, lang: data.language });
 
         telemetry.event(telemetryEventTypes.signup, collectPageParameters());
+
         const verifyOrGettingStarted = emailVerificationEnabled ? "auth/verify-email" : "getting-started";
-        const callBackUrl = `${
-          searchParams?.get("callbackUrl")
-            ? isOrgInviteByLink
-              ? `${WEBAPP_URL}/${searchParams.get("callbackUrl")}`
-              : addOrUpdateQueryParam(`${WEBAPP_URL}/${searchParams.get("callbackUrl")}`, "from", "signup")
-            : `${WEBAPP_URL}/${verifyOrGettingStarted}?from=signup`
-        }`;
+        const gettingStartedWithPlatform = "settings/platform/new";
+
+        const constructCallBackIfUrlPresent = () => {
+          if (isOrgInviteByLink) {
+            return `${WEBAPP_URL}/${searchParams.get("callbackUrl")}`;
+          }
+
+          return addOrUpdateQueryParam(`${WEBAPP_URL}/${searchParams.get("callbackUrl")}`, "from", "signup");
+        };
+
+        const constructCallBackIfUrlNotPresent = () => {
```

Comment:
not sure to understand the naming of this function, what does ifUrlNotPresent means

Reply:
I understand it's the callback url from searchParams but maybe we can find a better name or refactor to make stuff more clear

Reply:
`getUrlIfCallbackNotProvided`

Reply:
@ThyMinimalDev should we rename the functions to `constructCallBackUrlIfSearchParamsPresent` and `constructCallBackUrlIfSearchParamsNotPresent`?